pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -336,6 +381,64 @@ func isAdminCredSecret(namespace, secretName string) bool {
 	return false
 }
 
+// isMissingSecretLabel determines if the secret was createded by the CCO but has not been labelled yet
+func isMissingSecretLabel(secret metav1.Object) bool {
+	isAdmin := isAdminCredSecret(secret.GetNamespace(), secret.GetName())
+	_, hasAnnotation := secret.GetAnnotations()[minterv1.AnnotationCredentialsRequest]
+	value, hasLabel := secret.GetLabels()[minterv1.LabelCredentialsRequest]
+	hasValue := hasLabel && value == minterv1.LabelCredentialsRequestValue
+
+	return (isAdmin || hasAnnotation) && (!hasLabel || !hasValue)
+}
+
+type ReconcileSecretMissingLabel struct {
+	cachedClient   client.Client
+	mutatingClient corev1client.CoreV1Interface
+}
+
+var _ reconcile.Reconciler = &ReconcileSecretMissingLabel{}
+
+func (r *ReconcileSecretMissingLabel) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	start := time.Now()
+
+	logger := log.WithFields(log.Fields{
+		"controller": labelControllerName,
+		"secret":     fmt.Sprintf("%s/%s", request.NamespacedName.Namespace, request.NamespacedName.Name),
+	})
+
+	defer func() {
+		dur := time.Since(start)
+		metrics.MetricControllerReconcileTime.WithLabelValues(labelControllerName).Observe(dur.Seconds())
+	}()
+
+	logger.Info("syncing secret")
+	secret := &corev1.Secret{}
+	err := r.cachedClient.Get(ctx, request.NamespacedName, secret)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Debug("secret no longer exists")
+			return reconcile.Result{}, nil
+		}
+		logger.WithError(err).Error("error getting secret, requeuing")
+		return reconcile.Result{}, err
+	}
+
+	applyConfig := corev1applyconfigurations.Secret(secret.Name, secret.Namespace)
+	applyConfig.WithLabels(map[string]string{
+		minterv1.LabelCredentialsRequest: minterv1.LabelCredentialsRequestValue,
+	})
+
+	if _, err := r.mutatingClient.Secrets(secret.Namespace).Apply(ctx, applyConfig, metav1.ApplyOptions{
+		Force:        true, // we're the authoritative owner of this field and should not allow anyone to stomp it
```

Comment:
@deads2k would love a gut-check that this does what I think it does, given that the applyConfig is set up the way it is - is there a single owner for all of `metadata.labels`?

Reply:
Labels are divided by name, so you only own the labels that you're trying to set, not all the labels available.  At least that's my memory of it.  Is the reality different?